packages/vm/src/runBlock.ts

```diff
@@ -750,24 +752,24 @@ export async function rewardAccount(
 ): Promise<Account> {
   let account = await evm.stateManager.getAccount(address)
   if (account === undefined) {
-    if (common.isActivatedEIP(6800) === true) {
-      if (evm.verkleAccessWitness === undefined) {
+    if (common.isActivatedEIP(6800) === true && reward !== BIGINT_0) {
+      if (evm.systemVerkleAccessWitness === undefined) {
         throw Error(`verkleAccessWitness required if verkle (EIP-6800) is activated`)
       }
-      evm.verkleAccessWitness.readAccountHeader(address)
+      evm.systemVerkleAccessWitness.writeAccountHeader(address)
     }
     account = new Account()
   }
   account.balance += reward
   await evm.journal.putAccount(address, account)
 
-  if (common.isActivatedEIP(6800) === true) {
-    if (evm.verkleAccessWitness === undefined) {
+  if (common.isActivatedEIP(6800) === true && reward !== BIGINT_0) {
+    if (evm.systemVerkleAccessWitness === undefined) {
       throw Error(`verkleAccessWitness required if verkle (EIP-6800) is activated`)
     }
     // use vm utility to build access but the computed gas is not charged and hence free
-    evm.verkleAccessWitness.writeAccountBasicData(address)
-    evm.verkleAccessWitness.readAccountCodeHash(address)
+    evm.systemVerkleAccessWitness.writeAccountBasicData(address)
+    evm.systemVerkleAccessWitness.readAccountCodeHash(address)
```

Comment:
Can we consolidate all this down?  Instead of having a whole code block to check to write the account header when we have account non-existence and then a second check, can we just do (this is pseudo code but you get the idea)
```
let accountExists = true
if (account === undefined) {
    accountExists = false
    account = new Account()
}

if (6800 is active and reward > 0)
  accountExists ?  writeAccountData/readCodHash : writeAccountHeader
```